File: delivery_service/src/main/java/com/shichko/deliveryservice/model/dto/OrderDto.java
package com.shichko.deliveryservice.model.dto;

import lombok.Data;

import javax.persistence.Basic;
import javax.persistence.Column;
import javax.validation.Valid;
import javax.validation.constraints.Size;
import java.sql.Timestamp;
import java.util.Collection;

@Data
public class OrderDto extends AbstractDto {
    @Size(max = 255)
    private String address;
    @Size(max = 255)
    private String info;
    private String state;
    @Valid
    private CustomerDto customer;
    private Long courierId;
    private Collection<OrderedDto> ordered;
    private Timestamp orderedAt;
    private Timestamp deliveredAt;
    private Timestamp preferredRangeStart;
    private Timestamp preferredRangeEnd;
}

File: courier_application/app/src/main/java/by/bstu/vs/stpms/courier_application/model/entity/OrderDto.kt
package by.bstu.vs.stpms.courier_application.model.entity

import java.sql.Timestamp

class OrderDto (
    private val address: String,
    private val info: String? = null,
    private val state: String,
    private val customer: CustomerDto,
    private val courierId: Long? = null,
    private val ordered: Collection<OrderedDto>,
    private val orderedAt: Timestamp? = null,
    private val deliveredAt: Timestamp? = null
) : AbstractDto()